sonic_data_client/virtual_db.go

```diff
@@ -140,21 +203,35 @@ func v2rEthPortFieldStats(paths []string) ([]tablePath, error) {
 	var tblPaths []tablePath
 	if strings.HasSuffix(paths[KeyIdx], "*") {
 		for port, oid := range countersPortNameMap {
+			var oport string
+			if alias, ok := name2aliasMap[port]; ok {
+				oport = alias
+			} else {
+				log.V(2).Infof("%v dose not have a vendor alias", port)
+				oport = port
+			}
+
 			tblPath := tablePath{
-				dbName:       paths[DbIdx],
-				tableName:    paths[TblIdx],
-				tableKey:     oid,
-				field:        paths[FieldIdx],
-				delimitor:    separator,
-				jsonTableKey: port,
+				dbName:    paths[DbIdx],
+				tableName: paths[TblIdx],
+				tableKey:  oid,
+				field:     paths[FieldIdx],
+				delimitor: separator,
+				jsonTableKey: oport,
 				jsonField:    paths[FieldIdx],
 			}
 			tblPaths = append(tblPaths, tblPath)
 		}
 	} else { //single port
-		oid, ok := countersPortNameMap[paths[KeyIdx]]
+		var alias, name string
+		alias = paths[KeyIdx]
+		name = alias
+		if val, ok := alias2nameMap[alias]; ok {
+			name = val
+		}
+		oid, ok := countersPortNameMap[name]
 		if !ok {
-			return nil, fmt.Errorf(" %v not a valid port ", paths[KeyIdx])
+			return nil, fmt.Errorf(" %v not a valid sonic interface port, vendor alias is %v ", name, alias)
```

Comment:
Suggest rewording to remove redundancy: `"%v not a valid SONiC interface. Vendor alias is %v"`